spark_log_parser/parsing_models/application_model.py

```diff
@@ -121,79 +126,51 @@ def __init__(self, eventlogpath, bucket=None, stdoutpath=None, debug=False):  #
 
                     # print("Stage ids: %s" % stage_ids)
                     for stage_id in stage_ids:
-                        if stage_id not in self.jobs_for_stage:
-                            self.jobs_for_stage[stage_id] = [job_id]
-                        else:
-                            self.jobs_for_stage[stage_id].append(job_id)
+                        self.jobs_for_stage[stage_id].append(job_id)
 
                 elif event_type == "SparkListenerJobEnd":
                     job_id = json_data["Job ID"]
                     self.jobs[job_id].completion_time = json_data["Completion Time"] / 1000
                     self.jobs[job_id].result = json_data["Job Result"]["Result"]
+
                 elif event_type == "SparkListenerTaskEnd":
-                    stage_id = json_data["Stage ID"]
-                    # Add the event to all of the jobs that depend on the stage.
-                    for jid in self.jobs_for_stage[stage_id]:
-                        self.jobs[jid].add_event(json_data, True)
```

Comment:
Instead of making this Job --> Stage --> Task association eagerly, just construct our `TaskModel` and save it for later